common/experience.py

```diff
@@ -4,6 +4,67 @@
 from common.segment_tree import SumSegmentTree, MinSegmentTree
 
 
+class ReplayBufferCentralState(object):
+    def __init__(self, size):
+        """Create Replay buffer.
+        Parameters
+        ----------
+        size: int
+            Max number of transitions to store in the buffer. When the buffer
+            overflows the old memories are dropped.
+        """
+        self._storage = []
```

Comment:
I made measurements and found faster way to implement replay buffer with numpy only and preallocationg memory.
You can take a look into the common/experience.py